### PR TITLE
Added front-end topics validation

### PR DESCRIPTION
**Pull request by axifive**
- Added Front-End topics validation
- Updated Semantic UI

### CODE REVIEW

File: public/js/index.js [hunk 1]
```diff
@@ -2360,22 +2389,63 @@ function initTopicbar() {
             throttle: 500,
             cache: false,
             onResponse: function(res) {
-                var formattedResponse = {
+                let formattedResponse = {
```
Comment (techknowlogick) [inline public/js/index.js]: can you keep `var`? The versions of IE we support don't support newer ECMAscript

File: public/js/index.js [hunk 2]
```diff
@@ -2302,46 +2302,75 @@ function initNavbarContentToggle() {
 }
 
 function initTopicbar() {
-    var mgrBtn = $("#manage_topic")
-    var editDiv = $("#topic_edit")
-    var viewDiv = $("#repo-topic")
-    var saveBtn = $("#save_topic")
+    let mgrBtn = $("#manage_topic"),
```
Comment (techknowlogick) [inline public/js/index.js]: can you change this `let` to `var`?

File: public/js/index.js [hunk 3]
```diff
@@ -2302,46 +2302,75 @@ function initNavbarContentToggle() {
...
+                countPrompt: hidePrompt.children('#count_prompt').text(),
+                formatPrompt: hidePrompt.children('#format_prompt').text()
+            };
+        hidePrompt.remove();
+        return prompts;
+    }
 
     saveBtn.click(function() {
-        var topics = $("input[name=topics]").val();
+        let topics = $("input[name=topics]").val();
```
Comment (techknowlogick) [inline public/js/index.js]: same as above re: let/var

File: public/js/index.js [hunk 4]
```diff
@@ -2302,46 +2302,75 @@ function initNavbarContentToggle() {
...
-            } else {
+        }, function(data, textStatus, xhr){
+            if (xhr.responseJSON.status === 'ok') {
                 viewDiv.children(".topic").remove();
-                if (topics.length == 0) {
+                if (topics.length === 0) {
                     return
                 }
-                var topicArray = topics.split(",");
+                let topicArray = topics.split(",");
```
Comment (techknowlogick) [inline public/js/index.js]: same as above

File: public/js/index.js [hunk 5]
```diff
@@ -2302,46 +2302,75 @@ function initNavbarContentToggle() {
...
 
-                var last = viewDiv.children("a").last();
-                for (var i=0;i < topicArray.length; i++) {
-                    $('<div class="ui green basic label topic" style="cursor:pointer;">'+topicArray[i]+'</div>').insertBefore(last)
+                let last = viewDiv.children("a").last();
```
Comment (techknowlogick) [inline public/js/index.js]: same as above

File: public/js/index.js [hunk 6]
```diff
@@ -2302,46 +2302,75 @@ function initNavbarContentToggle() {
...
-            viewDiv.show();
-        }).fail(function(xhr) {
-            alert(xhr.responseJSON.message)
-        })
+        }).fail(function(xhr){
+            if (xhr.status === 422) {
+                if (xhr.responseJSON.invalidTopics.length > 0) {
+                    topicPrompts.formatPrompt = xhr.responseJSON.message;
+
+                    let invalidTopics = xhr.responseJSON.invalidTopics,
```
Comment (techknowlogick) [inline public/js/index.js]: same as above

**Review by techknowlogick — CHANGES_REQUESTED** [review]
Can you also change the permissions on ` public/vendor/plugins/semantic/semantic.min.css` back to 644?

**Comment by axifive** [comment]
@techknowlogick, Sure, missed it. What min version of IE are supported?

**Comment by techknowlogick** [comment]
@axifive thanks for making changes :) 

As for min IE, I think IE 11 is min, but let me ping @lafriks so he can say for sure.

**Comment by daviian** [comment]
@axifive @techknowlogick We directly infered the support from semantic ui. Did their support change with the version upgrade?

**Comment by axifive** [comment]
@daviian, No, still [supports IE 11+](https://github.com/Semantic-Org/Semantic-UI#browser-support)

File: templates/repo/home.tmpl [hunk 7]
```diff
@@ -28,22 +28,28 @@
 		{{if .IsRepositoryAdmin}}<a id="manage_topic" style="cursor:pointer;margin-left:10px;">{{.i18n.Tr "repo.topic.manage_topics"}}</a>{{end}}
 		</div>
 		{{if .IsRepositoryAdmin}}
-		<div class="ui repo-topic-edit grid" id="topic_edit" >
+		<div class="ui repo-topic-edit grid form segment error" id="topic_edit" >
 			<div class="fourteen wide column">
+				<div class="field">
 				<div class="ui fluid multiple search selection dropdown">
```
Comment (techknowlogick) [inline templates/repo/home.tmpl]: As there is now a additional div wrapping this div, can you indent?

**Comment by techknowlogick** [comment]
@axifive is the upgrade of semantic required for the form validation? I would prefer it not to upgrade semantic especially if this gets backported, in case the upgrade causes any issues.

**Comment by axifive** [comment]
@techknowlogick I tested the validation with 2.3.1 version because 2.2.1 has some problem  and throws the errors


2.2.1 was out two years ago and from that moment it was released many fixes and enhancements, I think we should update semantic-ui.

File: templates/repo/home.tmpl (hunk 7 above)
Comment (axifive) [reply templates/repo/home.tmpl]: Fixed

**Review by techknowlogick — APPROVED** [review]

**Comment by techknowlogick** [comment]
This LG-TM. @daviian as there is an upgrade to semantic UI I don't think we should backport it to 1.5 in case it causes bugs. That is not a strong opinion of mine, and so I'll leave the label as is.

**Review by lafriks — APPROVED** [review]

**Pull request merged**